### PR TITLE
Fix OptimizerHacks for clang 17

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Utilities/OptimizerHacks.cpp [hunk 1]
```diff
@@ -9,7 +9,8 @@ void indicate_value_can_be_changed(void* /*variable*/) {}
 }  // namespace optimizer_hacks
 #endif  /* defined(__clang__) && __clang__ < 11 */
 
-#if defined(__clang__) && __clang_major__ >= 15 && defined(__GLIBCXX__)
+#if defined(__clang__) && __clang_major__ >= 15 \
+  && __clang_major__ < 17 && defined(__GLIBCXX__)
 #include <string>
 // clang v15+ fails to instantiate the following std library templates if
```
Comment (knelli2) [inline src/Utilities/OptimizerHacks.cpp]: Update code comment as well?

**Review by knelli2 — CHANGES_REQUESTED** [review]

**Review by wthrowe — CHANGES_REQUESTED** [review]
It works with clang 16 without the hack now, so this is probably a change in SpECTRE, not a change in clang.  I'd delete the whole section and see if it breaks for anyone.

Edit: Clang 15 fails to compile for unrelated reasons.  (Duplicate `Composition` instantiations)

**Comment by nilsdeppe** [comment]
Okay, deleted the code block. Thanks for reviewing!

**Review by wthrowe — APPROVED** [review]

**Comment by kidder** [comment]
works for me

**Review by knelli2 — APPROVED** [review]
Good enough for me

**Pull request merged**